### PR TITLE
small refactor to shorten the lock on slot_under_contention hashset

### PR DESCRIPTION
**Pull request by HaoranYi**
#### Summary of Changes

Fixes #

### CODE REVIEW

File: runtime/src/accounts_db.rs [hunk 1]
```diff
@@ -4853,22 +4853,13 @@ impl AccountsDb {
...
-            if slots_under_contention.contains(&slot) {
-                true
-            } else {
-                slots_under_contention.insert(slot);
-                false
-            }
-        };
-
-        if !is_being_purged {
+        if self
```
Comment (jeffwashington) [inline runtime/src/accounts_db.rs:4856]: I suggest either a comment about this result means `!is_being_purged` or a local of `is_being_purged` like existed before and checking if ! again. I don't know enough from the name of the struct `slots_under_contention` to associate `!is_being_purged`

Comment (HaoranYi) [reply runtime/src/accounts_db.rs:4856]: Yeah. Added the comments.

**Review by jeffwashington — DISMISSED** [review]
lgtm

**Review by carllin — APPROVED** [review]
Thanks for the cleanup!

**Pull request merged**